runner_service/controllers/login.py

```diff
@@ -0,0 +1,80 @@
+from flask_restful import request
+import logging
+from .utils import requires_auth, log_request
+from .base import BaseResource
+from ..services.utils import APIResponse
+from runner_service import configuration
+import jwt
+import datetime
+
+logger = logging.getLogger(__name__)
+
+
+class Login(BaseResource):
+    """Authenticate user and provide token"""
+
+    @requires_auth
+    @log_request(logger)
+    def get(self):
+        """
+        GET {username, password}
+        Login to get token
+
+        Example.
+
+        ```
+        $ curl -k -i -H "Content-Type: application/json" --data '{"username":"admin", "password": "admin"}' https://localhost:5001/api/v1/login -X get
```

Comment:
Normally a username and password request adopts  basic auth in the http header - e..g curl --user username:password not a json payload. Although this works, I think it goes against standards

Reply:
+1